service/src/main/java/bio/terra/catalog/rawls/RawlsService.java

```diff
@@ -81,6 +85,76 @@ WorkspacesApi workspacesApi(AuthenticatedUserRequest user) {
     return new WorkspacesApi(getApiClient(user));
   }
 
+  //  public List<Entity> entityList(AuthenticatedUserRequest user, String storageSourceId)  {
+  //    WorkspaceResponse response = null;
+  //    try {
+  //      response = workspacesApi(user).getWorkspaceById(storageSourceId, List.of());
+  //      var entities = entitiesApi(user).listEntities(response.getWorkspace().getNamespace(),
+  // response.getWorkspace().getName(), Sample);
+  //      return entities;
+  //    } catch (ApiException e) {
+  //      throw new RuntimeException(e);
+  //    }
+  //  }
+
+  public EntityQueryResponse entityQuery(
+      AuthenticatedUserRequest user, String storageSourceId, String tableName) {
+    WorkspaceResponse response;
+    try {
+      response = workspacesApi(user).getWorkspaceById(storageSourceId, List.of());
+      return entitiesApi(user)
+          .entityQuery(
+              response.getWorkspace().getNamespace(),
+              response.getWorkspace().getName(),
+              tableName,
+              null,
+              null,
+              null,
+              null,
+              null,
+              List.of(),
+              null,
+              null);
+    } catch (ApiException e) {
+      throw new RawlsException("Entity Query Failed", e);
+    }
+  }
+
+  public Map<String, EntityTypeMetadata> entityMetadata(
+      AuthenticatedUserRequest user, String storageSourceId) {
+    WorkspaceResponse response;
+    try {
+      response = workspacesApi(user).getWorkspaceById(storageSourceId, List.of());
+      var entities =
+          entitiesApi(user)
+              .entityTypeMetadata(
+                  response.getWorkspace().getNamespace(),
+                  response.getWorkspace().getName(),
+                  true,
+                  null);
+      //noinspection unchecked
+      return ((Map<String, Map<String, Object>>) entities)
+          .entrySet().stream()
+              .collect(
+                  Collectors.toMap(
+                      Map.Entry::getKey,
+                      entry -> {
+                        Map<String, Object> value = entry.getValue();
+                        //noinspection unchecked
+                        return new EntityTypeMetadata()
+                            .idName((String) value.get("idName"))
+                            .count((Integer) value.get("count"))
+                            .attributeNames((List<String>) value.get("attributeNames"));
+                      }));
+    } catch (ApiException e) {
+      throw new RawlsException("Entity Metadata failed", e);
+    }
+  }
+
+  EntitiesApi entitiesApi(AuthenticatedUserRequest user) {
+    return new EntitiesApi(getApiClient(user));
+  }
```

Comment:
When you rebase you should be able to pick up this change from my code.

Reply:
will do